Python/Vault/pyVaultNode.h

```diff
@@ -0,0 +1,92 @@
+/* This file is part of HSPlasma.
+ *
+ * HSPlasma is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * HSPlasma is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with HSPlasma.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _PYVAULTNODE_H
+#define _PYVAULTNODE_H
+
+#include "PyPlasma.h"
+#include <Vault/plVaultNode.h>
+#include <Vault/plVaultNodeTypes.h>
+
+extern PyTypeObject pyVault_Type;
+PyObject* Init_pyVault_Type();
+
+PY_WRAP_PLASMA_VALUE(VaultNode, plVaultNode);
+
+PY_WRAP_PLASMA_VALUE(VaultPlayerNode, plVaultPlayerNode);
+PY_WRAP_PLASMA_VALUE(VaultAgeNode, plVaultAgeNode);
+PY_WRAP_PLASMA_VALUE(VaultGameServerNode, plVaultGameServerNode);
+PY_WRAP_PLASMA_VALUE(VaultAdminNode, plVaultAdminNode);
+PY_WRAP_PLASMA_VALUE(VaultServerNode, plVaultServerNode);
+PY_WRAP_PLASMA_VALUE(VaultFolderNode, plVaultFolderNode);
+PY_WRAP_PLASMA_VALUE(VaultPlayerInfoNode, plVaultPlayerInfoNode);
+PY_WRAP_PLASMA_VALUE(VaultSystemNode, plVaultSystemNode);
+PY_WRAP_PLASMA_VALUE(VaultImageNode, plVaultImageNode);
+PY_WRAP_PLASMA_VALUE(VaultTextNoteNode, plVaultTextNoteNode);
+PY_WRAP_PLASMA_VALUE(VaultSDLNode, plVaultSDLNode);
+PY_WRAP_PLASMA_VALUE(VaultAgeLinkNode, plVaultAgeLinkNode);
+PY_WRAP_PLASMA_VALUE(VaultChronicleNode, plVaultChronicleNode);
+PY_WRAP_PLASMA_VALUE(VaultPlayerInfoListNode, plVaultPlayerInfoListNode);
+PY_WRAP_PLASMA_VALUE(VaultMarkerNode, plVaultMarkerNode);
+PY_WRAP_PLASMA_VALUE(VaultAgeInfoNode, plVaultAgeInfoNode);
+PY_WRAP_PLASMA_VALUE(VaultAgeInfoListNode, plVaultAgeInfoListNode);
+PY_WRAP_PLASMA_VALUE(VaultMarkerListNode, plVaultMarkerListNode);
+
+inline PyObject* pyPlasma_convert(const plVaultNode& node)
+{
+    if (plVaultPlayerNode* access = node.upcastToPlayerNode())
```

Comment:
Each of these upcast calls checks fNodeType independently, making this somewhat inefficient...  Why not use a switch instead?

Reply:
Good point. This was mostly me being lazy, not wanting to have to bang out the cast myself, due to the scope of the changes. I'll go back and hit this.

Reply:
Updated to fix this